factcast-site/documentation/content/setup/server/tail-index.md

```diff
@@ -0,0 +1,116 @@
++++
+title = "Tail Index"
+
+creatordisplayname = "Maik Toepfer"
+creatoremail = "maik.toepfer@prisma-capacity.eu"
+
+parent = "usage"
+identifier = "tail-index"
+weight = 220
++++
+
+The *tail index* is a performance optimization for FactCast which speeds up queries 
+against the end (the *tail*) of the fact log.
+
+Background
+----------
+
+### Global Index
+
+FactCast uses a Postgres database for its persistence. Events (or *Facts*) live in a single database table
+called *fact* which is referred to as the *fact log*. To speed up access to the fact log, 
+a [global index](https://www.postgresql.org/docs/11/textsearch-indexes.html) is used. 
+However, as the fact log is constantly growing, so is the index. 
+With the global index alone, query performance decreases.
+
+
+### Subscription Phases
+
+Generally, a [subscription]({{< ref "concept/_index.md#read-subscribe" >}}) consists of two phases:
+1. Catching up with past events, starting from the beginning of the fact log
+2. Checking for new events by querying the tail of the fact log
+
+While 1.) is happening only once at the beginning of a subscription, 2.) is happening regularly. 
+For example, a [Factus managed projection]({{< ref "managed-projection.md">}}) is updated via a call to `factus.update(myProjection)`.
+The first call is expensive as it is catching up with all past events (1.). Subsequent calls of `update()`
+only ask for new events at the tail of the fact log (2.)   
+
+
+Tail Index
+----------
+
+A *tail index* supports the regular "Are there new events?" queries by placing one or more indexes 
+at *the end* of the fact log:
+
+![](../tail-index.png)
+
+More precisely, FactCast maintains a certain number of rolling tail indexes. Once an index becomes too old,
+FactCast removes it and creates a new smaller one. When asked to query facts from the end of the fact log, 
+the Postgres database has now multiple options. Beside the large global index, there are now much smaller ones
+which most likely cover the query and are cheaper to access.  
+
+For further details on how to configure tail indexes, see [the configuration properties]({{< ref "properties.md#performance--reliability" >}}). 
+
+Note: Tail indexes are implemented as [Postgres Partial Indexes](https://www.postgresql.org/docs/11/indexes-partial.html). 
+
+
+Index Maintenance Trade-Off
```

Comment:
ok, I assume you mean the "fast update" feature of the GIN index: https://www.postgresql.org/docs/11/gin-implementation.html#GIN-FAST-UPDATE

What I understand: 
- a tail index is created with fastupdate=on as this is the default
- fastupdate postpones the update of the GIN index since it first populates a temporary list of changes (the "pending list")
- under the hood, an index search uses both the actual index and the temporary list
- merging the pending list can is influenced by multiple factors, e.g. vacuum or autoanalyze of the fact table. Also, there is a max threshold `gin_pending_list_limit` and a special function.

here is what I've added (plus some links):
A tail index is a Postgres GIN index with enabled fastupdate(link). If you encounter performance issues, see the Postgres documentation(link) for further advice.